pkg/utils/parse_arf_result_test.go

```diff
@@ -166,6 +166,54 @@ var _ = Describe("XCCDF parser", func() {
 
 	})
 
+	Describe("Test for Check Result Variable Association ", func() {
+		BeforeEach(func() {
+			mcInstance := &mcfgv1.MachineConfig{}
+			schema = scheme.Scheme
+			schema.AddKnownTypes(mcfgv1.SchemeGroupVersion, mcInstance)
+			resultsFilename = "../../tests/data/xccdf-result-remdiation-templating.xml"
+			dsFilename = "../../tests/data/ds-input-for-remediation-value.xml"
+		})
+
+		JustBeforeEach(func() {
+			xccdf, err = os.Open(resultsFilename)
+			Expect(err).NotTo(HaveOccurred())
+
+			ds, err = os.Open(dsFilename)
+			Expect(err).NotTo(HaveOccurred())
+			dsDom, err := ParseContent(ds)
+			Expect(err).NotTo(HaveOccurred())
+			resultList, err = ParseResultsFromContentAndXccdf(schema, "testScan", "testNamespace", dsDom, xccdf)
+			Expect(resultList).NotTo(BeEmpty())
+			nChecks, nRems = countResultItems(resultList)
+		})
+
+		Context("Check if the check result has correct value used attribute", func() {
+			const (
+				expCheckResultName = "testScan-auditd-data-retention-max-log-file"
+				expValue           = "var-auditd-max-log-file"
+			)
+
+			var (
+				check *compv1alpha1.ComplianceCheckResult
+			)
+
+			BeforeEach(func() {
+				for i := range resultList {
+					if resultList[i].CheckResult != nil && resultList[i].CheckResult.Name == expCheckResultName {
+						check = resultList[i].CheckResult
+						break
+					}
+				}
+			})
+
+			It("Should have the expected value List", func() {
+				Expect(len(check.ValueUsed)).To(Equal(1))
+				Expect(check.ValueUsed[0]).To(Equal(expValue))
+			})
+		})
```

Comment:
Let's do several tests, one for a rule where you know there's no values used, then one that has several values (more than one). I think that would be enough coverage.

Reply:
I have added two more tests, one with 2 variables and one without any